### ISSUE TITLE
IP from dnsjava not matching dig.

### ISSUE THREAD

**Issue by Dmole** [dnsjava/dnsjava#214]
After using setResolver ARecord.rdataToString() shows the private IP instead of the public IP returned from the DNS server (checked with `dig @server query +short`).
Is there a way to disable this behavior?
I believe this is a regression bug.

**Comment by ibauersachs** [comment]
I don't really get what you're doing. Can you please post some sample code that shows the bug?

**Comment by Dmole** [comment]
sample code:

```
# bash
dig @dns.eg.com eg2.eg.com +short
# eg.eg.com.
# 127.a.piblic.ipv4
```

```
// Java
// should be a CNAME but we don't know that yet
Lookup lookup = new Lookup("eg2.eg.com", Type.A);
lookup.setResolver(new SimpleResolver("dns.eg.com"));
ARecord a = (ARecord) (lookup.run()[0]);
System.out.println(a.rdataToString());
// eg.eg.com. 0 IN A 127.a.NATed.ipv4
```

**Comment by Dmole** [comment]
Note that to see the issue you need a DNS server with an A record pointing to a public IP that is NATed to a private IP. I just ran the code above and found the issue with Amazon elastic IPs.
put the private IP in /etc/hosts and watch the resolver get ignored.
```
java -cp slf4j-api-1.7.30.jar:dnsjava-3.4.1.jar:. Test dns.eg.com eg.eg.com
```

**Comment by Dmole** [comment]
Adding 
```
lookup.setCache(null);
lookup.setDefaultHostsFileParser(null);
```
did not fix the issue, is there a way to force setResolver to be respected?

**Comment by Dmole** [comment]
```
System.setProperty("dnsjava.lookup.use_hosts_file", "false")
```
solves the issue... but that's really not good to use a global flag to make an instance method actually work.
Would dnsjava be open to fixing/changing this behavior?

**Comment by ibauersachs** [comment]
You need to call `Lookup.setDefaultHostsFileParser(null);` _before_ creating a `Lookup` instance OR call `lookup.setHostsFileParser(null)`. I'm not sure what exactly you're missing: you can set a default, and you can override the default per instance.

**Comment by Dmole** [comment]
Thanks, that helps.
Would it be reasonable to have setResolver call setHostsFileParser(null) and/or add a warning to the javadoc?

**Comment by ibauersachs** [comment]
No, this is intended behavior. The `Resolver` implementations talk to DNS servers, while `Lookup` (and `LookupSession`), well, does lookups just like the OS or C's `gethostbyname` does.

**Comment by Dmole** [comment]
Anyway the next person who upgrades from an old version will hopefully find this issue (or do a better job finding it themselves than I did). Thanks the solution.

**Comment by ibauersachs** [comment]
It's mentioned in the release notes `- Add support for hosts file in lookups (#195)`, the config property is listed in the readme and the two API methods are IMO quite obvious. I don't know what else I could have done.

**Comment by Dmole** [comment]
As I said adding a line to the setResolver javadoc was what I was expecting;

> Warning unless setHostsFileParser(null) is setResolver may be ignored.
